### PR TITLE
Add step to record image-registry generation number

### PR DESCRIPTION
**Pull request by wzheng1**
@openshift/devexp-qe Please help to review, thanks!

### CODE REVIEW

File: features/step_definitions/registry.rb [hunk 1]
```diff
@@ -385,8 +386,18 @@
...
 Given /^default image registry route is stored in the#{OPT_SYM} clipboard$/ do |cb_name| 
+  ensure_admin_tagged
+  ensure_destructive_tagged
   org_proj_name = project(generate: false).name rescue nil
   cb_name ||= :registry_route
   cb[cb_name] = route('default-route', service('default-route',project('openshift-image-registry'))).dns(by: admin)
   project(org_proj_name) if org_proj_name
 end
+
+Given /^current generation number of image-registry deployment is stored into#{OPT_SYM} clipboard$/ do |cb_name|
```
Comment (xiuwang) [inline features/step_definitions/registry.rb]: Could we make this more suitable for other resources?

Comment (wzheng1) [reply features/step_definitions/registry.rb]: Good suggestion!

Comment (wzheng1) [reply features/step_definitions/registry.rb]: Done, please check.

**Comment by xiuwang** [comment]
LGTM, thanks

File: features/step_definitions/registry.rb [hunk 2]
```diff
@@ -385,8 +386,18 @@
...
 Given /^default image registry route is stored in the#{OPT_SYM} clipboard$/ do |cb_name| 
+  ensure_admin_tagged
+  ensure_destructive_tagged
   org_proj_name = project(generate: false).name rescue nil
   cb_name ||= :registry_route
   cb[cb_name] = route('default-route', service('default-route',project('openshift-image-registry'))).dns(by: admin)
   project(org_proj_name) if org_proj_name
 end
+
+Given /^current generation number of#{OPT_QUOTED} deployment under #{QUOTED} is stored into#{OPT_SYM} clipboard$/ do |name,namespace,cb_name|
```
Comment (pruan-rht) [inline features/step_definitions/registry.rb]: @wzheng1, it looks like the name of the deployment is required?  If so then  `#{OPT_QUOTED}` should be `#{QUOTED}`.

Comment (pruan-rht) [reply features/step_definitions/registry.rb]: @wzheng1 perhaps we should add a method to `lib/openshift/deployment.rb` to support getting the `generation` field.

Comment (pruan-rht) [reply features/step_definitions/registry.rb]: @wzheng1 added helper with this PR https://github.com/openshift/verification-tests/pull/315

Comment (wzheng1) [reply features/step_definitions/registry.rb]: @pruan-rht Thanks for the pr, it simplifies my definition!

File: features/step_definitions/registry.rb [hunk 3]
```diff
@@ -385,8 +386,16 @@
...
+  ensure_admin_tagged
+  ensure_destructive_tagged
   org_proj_name = project(generate: false).name rescue nil
   cb_name ||= :registry_route
   cb[cb_name] = route('default-route', service('default-route',project('openshift-image-registry'))).dns(by: admin)
   project(org_proj_name) if org_proj_name
 end
+
+Given /^current generation number of#{OPT_QUOTED} deployment is stored into#{OPT_SYM} clipboard$/ do |name, cb_name|
+  ensure_admin_tagged
```
Comment (akostadinov) [inline features/step_definitions/registry.rb]: getting `deployment` info should not require `admin` tag. Maybe for some deployments. But we will already be admins.

File: features/step_definitions/registry.rb [hunk 4]
```diff
@@ -385,8 +386,16 @@
 end
 
 Given /^default image registry route is stored in the#{OPT_SYM} clipboard$/ do |cb_name| 
+  ensure_admin_tagged
+  ensure_destructive_tagged
```
Comment (akostadinov) [inline features/step_definitions/registry.rb]: I don't think this operation is destructive or can interfere with other operations. Can we remove the `ensure_destructive` call?

File: features/step_definitions/registry.rb (hunk 3 above)
Comment (wzheng1) [reply features/step_definitions/registry.rb]: Yeah, you are correct. Tag is removed.

File: features/step_definitions/registry.rb (hunk 4 above)
Comment (wzheng1) [reply features/step_definitions/registry.rb]: Yeah, you are correct. Tag is removed.

**Pull request merged**